include/maidsafe/crux/socket.hpp

```diff
@@ -250,6 +250,7 @@ inline void socket::close() {
     // Already closed?
     if (!multiplexer) return;
 
+    keepalive_timer.stop();
```

Comment:
I have hoped to address the close packet as a whole some time in future. Right now, not having this line there was causing io_service::run function to block even when the socket was already destroyed.

In future, I would imagine at this point a shared (say) active_close state would be created, the keepalive_timer together with the multiplexer would be moved into it and the timers wait time would be adjusted to some small value.

(FYI: I fixed the comment above)